From 69714573dcdffd4e5001ae246c665078a60ee37b Mon Sep 17 00:00:00 2001
From: Jan Stancek <jstancek@redhat.com>
Date: Thu, 19 May 2016 16:43:03 +0200
Subject: [PATCH] mem/oom: fork may fail before test

On systems with small RAM and vm.overcommit_memory == 2,
the system may already be in state where Committed_AS is
larger than CommitLimit due to various backround daemons
(NM, tuned, polkitd,...) asking for big allocations.

In such instances oom testcases report TBROK on very first fork.
This patch allows this to be a valid outcome (TPASS).

Signed-off-by: Jan Stancek <jstancek@redhat.com>
Acked-by: Cyril Hrubis <chrubis@suse.cz>
---
 testcases/kernel/mem/lib/mem.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/testcases/kernel/mem/lib/mem.c b/testcases/kernel/mem/lib/mem.c
index c55a55d..3d853a3 100644
--- a/testcases/kernel/mem/lib/mem.c
+++ b/testcases/kernel/mem/lib/mem.c
@@ -127,6 +127,10 @@ void oom(int testcase, int lite, int retcode, int allow_sigkill)
 
 	switch (pid = fork()) {
 	case -1:
+		if (errno == retcode) {
+			tst_resm(TPASS | TERRNO, "fork");
+			return;
+		}
 		tst_brkm(TBROK | TERRNO, cleanup, "fork");
 	case 0:
 		threads = MAX(1, tst_ncpus() - 1);
-- 
1.8.3.1

